tensorflow/python/kernel_tests/xent_op_deterministic_test.py

```diff
@@ -94,39 +104,66 @@ def _generateInputs(self, dtype, seed=123):
 
   @test_util.run_in_graph_and_eager_modes
   def testForward(self):
-    with self.session(), test_util.force_cpu():
+    with self.cached_session(), test_util.device(use_gpu=True):
```

Comment:
I think `test_util.device` is unnecessary, since the test will use the GPU by default if available.

Reply:
I agree, and I've thought about this. `use_gpu=True` explicitly pins the ops to the GPU, if there is a GPU. I understand that the expected, default behavior is for the ops to auto-pin to the GPU (if there is one), but I've wondered if we should rely on that. Considering again ...

Reply:
The next commit will remove almost all instances of `use_gpu=True`, in `xent_op_test_base.py`, `xent_op_test.py`, and `xent_op_deterministic_test.py`.